torch/fx/passes/infra/partitioner.py

```diff
@@ -205,8 +205,8 @@ def _update_partition_map(node: Node, id: int):
                 merge_candidates[partition_id] = None
 
             # merge all possible partitions
-            for node in assignment:
-                merge_candidates[assignment[node]] = None
+            for partition_id in partitions_by_id:
+                merge_candidates[partition_id] = None
```

Comment:
```suggestion
            merge_candidates.update(dict.fromkeys(partitions_by_id))
```
may be marginally faster

Reply:
I tested these two versions:
```
import time
cur = time.time()
for i in range(10000):
    a = {}
    b = {}
    for j in range(10):
        a[i + j] = j
    # b.update(dict.fromkeys(a))
    for key in a:
        b[key] = None
print(time.time() - cur)
```
The result show that for-loop is a little faster than dict.fromkeys. Maybe dict.fromkeys is faster than for-loop if the size of ```a``` is large enough? But I think we can assume the number of partitions is small.

Reply:
@lingzhi98 we have various internal use cases where the number of partitions generated are >100 in some cases. If `b.update(dict.fromkeys(a)` is faster for > 10 elements do you mind doing this change? Other than that this PR looks good.

Reply:
Done.